Quick-Start/Prerequisite.md

```diff
@@ -1,101 +1,101 @@
+    - Extensions are installed using the SQL command `CREATE EXTENSION`, and updated with the `ALTER EXTENSION` command, both of which reference the extension's control file. 
+    - During installation, PostgreSQL reads the control file, validates dependencies, executes the installation script, and sets up the extension in the database. 
+    - Extensions can be managed using SQL commands to enable, disable, upgrade, or uninstall them. 
+
+6.  **Extension Upgrades:** 
+    - The extension control file specifies the version of the extension, allowing PostgreSQL to manage upgrades seamlessly. 
+    - When a new version of an extension is installed, PostgreSQL compares the version specified in the control file with the installed version. 
+    - If an upgrade script is provided, it is executed to migrate data or perform other necessary changes during the upgrade process. 
+
+The extension architecture provides a structured and standardized way to extend PostgreSQL's functionality. It allows developers to create reusable components, customize the database to specific needs, and integrate third-party functionality seamlessly. The modular design and integration with the Extension API ensure compatibility, ease of installation, and manageability of extensions within the PostgreSQL ecosystem. 
+
+# Supported Languages: 
+PostgreSQL supports various target languages for extension development. Here is a list of the supported target languages in Postgres Extension Creation: 
+
+## C: 
+The C programming language is widely used for developing PostgreSQL extensions. It offers direct access to PostgreSQL internals and provides high performance. 
+ - Development Environment: To develop C-based extensions for PostgreSQL, you need a C compiler (such as GCC) and the necessary development libraries for PostgreSQL.   
+   - On Linux systems, you may need to install the `postgresql-server-dev` package.  
+   - On macOS, you can use Xcode Command Line Tools or install PostgreSQL via Homebrew.  
+   - On Windows, you can use Microsoft Visual Studio or MinGW. 
+ - Build System: C-based extensions typically use Makefiles or build systems like CMake or Autotools to compile and link the extension code. 
+
+## C++:  
+PostgreSQL also supports C++ for extension development. C++ allows object-oriented programming and provides additional features over the C language. 
+ - Development Environment: Similar to C, you need a C++ compiler and the development libraries for PostgreSQL. 
+ - Build System: `C++` extensions can use Makefiles or build systems like CMake or Autotools for compilation and linking. 
+
+## PL/pgSQL: 
+`PL/pgSQL` is the default procedural language of PostgreSQL. It is similar to `SQL` and allows you to write stored procedures, functions, and triggers using SQL-like syntax. 
+ - Development Environment: No additional development environment is required. You can write `PL/pgSQL` functions directly in the PostgreSQL database using SQL client tools. 
```

Comment:
Yes, but ultimately you need to save your SQL to a file to be installed by the extension.